### PR TITLE
Feature/discord oauth p2 - functionality

### PR DESCRIPTION
**Pull request by cherylli**
# Description

Add discord login functionality, go to http://localhost:8000/api/v1/auth/discord/login, a discord login / authorize screen will pop up, click authorize and it will grant access token, refresh token and then redirect to the frontend url

It gets the user discord email, 
- if this email exists in our database, we add a discord oauth profile to that user
- if the email does not exist, we create a new user + an oauth profile

Currently, if the user has different emails for credentials, and differnet oauth, it will create different accounts as we have no way to match users with different emails. In future, we should make it so user can link their social accounts to their existing email, this way, we can link their accounts without needing to match their emails

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature updates / changes
- [ ] Tests
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Login, and check access_token, refresh_token in cookies
Make sure to db seed before testing, discord needs to be in the oauthProvider table otherwise it will generate an error

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: prisma/schema/user.prisma [hunk 1]
```diff
@@ -65,7 +65,7 @@ model UserRole {
 
 model OAuthProvider {
   id   Int    @id @default(autoincrement())
-  name String
+  name String @unique
```
Comment (cherylli) [inline prisma/schema/user.prisma:68]: I have generated a migrate for this in #190, so I won't be doing it here

**Review by timDeHof — DISMISSED** [review]
All unit and e2e test pass and 0authProvider in the database shows discord in the model

**Comment by cherylli** [comment]
@timDeHof need a re-review if you can please, resolved changelog conflict

**Review by Ajen07 — APPROVED** [review]
The Discord login flow is working well, and new user is being added to the database correctly.

**Review by timDeHof — APPROVED** [review]
Every unit & e2e tests passed. Also able to login into the dashboard using discord

**Pull request merged**